geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/RasterRDD.scala

```diff
@@ -90,74 +103,86 @@ abstract class RasterRDD[K: AvroRecordCodec: ClassTag] extends TileRDD[K] {
+    (TileRDD.getCRS(crs), layout) match {
+      case (Some(crs), Right(layoutDefinition)) =>
+          rdd.collectMetadata[SpatialKey](crs, layoutDefinition)
+      case (None, Right(layoutDefinition)) =>
+          rdd.collectMetadata[SpatialKey](layoutDefinition)
+      case (Some(crs), Left(layoutScheme)) =>
+          rdd.collectMetadata[SpatialKey](crs, layoutScheme)._2
+      case (None, Left(layoutScheme)) =>
+          rdd.collectMetadata[SpatialKey](layoutScheme)._2
+    }
   }.toJson.compactPrint
 
   def cutTiles(layerMetadata: String, resampleMethod: String): TiledRasterRDD[SpatialKey] = {
     val md = layerMetadata.parseJson.convertTo[TileLayerMetadata[SpatialKey]]
     val rm = TileRDD.getResampleMethod(resampleMethod)
-    new SpatialTiledRasterRDD(MultibandTileLayerRDD(rdd.cutTiles(md, rm), md))
+    new SpatialTiledRasterRDD(None, MultibandTileLayerRDD(rdd.cutTiles(md, rm), md))
   }
 
   def tileToLayout(tileLayerMetadata: String, resampleMethod: String): TiledRasterRDD[SpatialKey] = {
     val md = tileLayerMetadata.parseJson.convertTo[TileLayerMetadata[SpatialKey]]
     val rm = TileRDD.getResampleMethod(resampleMethod)
-    new SpatialTiledRasterRDD(MultibandTileLayerRDD(rdd.tileToLayout(md, rm), md))
+    new SpatialTiledRasterRDD(None, MultibandTileLayerRDD(rdd.tileToLayout(md, rm), md))
   }
 
-  def reproject(target_crs: String, resampleMethod: String): ProjectedRasterRDD = {
-    val tcrs = TileRDD.getCRS(target_crs).get
+  def reproject(targetCRS: String, resampleMethod: String): ProjectedRasterRDD = {
+    val crs = TileRDD.getCRS(targetCRS).get
     val resample = TileRDD.getResampleMethod(resampleMethod)
-    new ProjectedRasterRDD(rdd.reproject(tcrs, resample))
+    new ProjectedRasterRDD(rdd.reproject(crs, resample))
   }
 
 }
 
-class TemporalProjectedRasterRDD(val rdd: RDD[(TemporalProjectedExtent, MultibandTile)]) extends RasterRDD[TemporalProjectedExtent] {
-
-  def collectMetadata(crs: Option[CRS], extent: Option[Extent], layout: Option[TileLayout], tileSize: Int): String = {
-      (crs, extent, layout) match {
-        case (Some(crs), Some(extent), Some(layout)) =>
-          rdd.collectMetadata[SpaceTimeKey](crs, LayoutDefinition(extent, layout))
-        case (None, Some(extent), Some(layout)) =>
-          rdd.collectMetadata[SpaceTimeKey](LayoutDefinition(extent, layout))
-        case (Some(crs), _, _) =>
-          rdd.collectMetadata[SpaceTimeKey](crs, FloatingLayoutScheme(tileSize, tileSize))._2
-        case (None, None, None) =>
-          rdd.collectMetadata[SpaceTimeKey](FloatingLayoutScheme())._2
-        case _ =>
-          throw new IllegalArgumentException(s"Can't handle $crs, $extent, $layout, $tileSize")
-      }
+class TemporalRasterRDD(val rdd: RDD[(TemporalProjectedExtent, MultibandTile)]) extends RasterRDD[TemporalProjectedExtent] {
+
+  def collectMetadata(layout: Either[LayoutScheme, LayoutDefinition], crs: String): String = {
```

Comment:
Lets make scala interfaces `protected`, which should help mark them as not for external consumption